## 92% of end-end tests have passed
```
Total executed tests: 24
Failed tests: 2
Expected to fail tests: 0
Passed tests: 22
```
```
IDs of failed tests: 741612,741968 
```
<h3>Failed tests (2)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=21590-fix-group-member-limit&PR_ID=22561&APK_NAME=22561.apk&TR_CASE_IDS=741612,741968">Rerun failed tests</a></li><br/><h4>Class TestProfileMultipleDevices:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3858742">test_profile_change_username</a>, id: 741968 </b></td></tr><tr><td><p><blockquote><div>Device 1: Tap on found: LogInButton </div><div>## Signed in successfully!</div></blockquote></p>

```
profile/test_profile.py:180: in test_profile_change_username
    self.errors.verify_no_errors()
base_test_case.py:207: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Device 2: Updated username 'user 1' is not shown in the community channel
E    Device 2: Updated username is not shown in the mentions list
E    Device 2: Updated username is absent in the contacts list
E    Device 2: Can't find chat with updated username
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387432540875CFB">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387165491605SPQ">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3858738">test_wallet_bridge_flow_mainnet</a>, id: 741612 </b></td></tr><tr><td><p><blockquote><div># STEP: Checking bridge from Base to Optimism</div><div>Device 1: Find `Button` by `xpath`: `//*[@content-desc='token-network']/android.widget.TextView[@text='Ether']` </div></blockquote></p>

```
wallet/test_wallet_mainnet_no_send.py:254: in test_wallet_bridge_flow_mainnet
    self.wallet_view.select_asset('Ether')
../views/wallet_view.py:305: in select_asset
    xpath="//*[@content-desc='token-network']/android.widget.TextView[@text='%s']" % asset_name).click()
../views/base_element.py:90: in click
    element = self.find_element()
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 1: Button by xpath: `//*[@content-desc='token-network']/android.widget.TextView[@text='Ether']` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387560528028DDO">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Passed tests (22)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3858726">test_group_chat_join_send_text_messages_push</a>, id: 702807 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387556605314IQE">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387260855357RTE">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387152203575IGR">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletCustomParamOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3858746">test_send_snt_custom_tx_params</a>, id: 742910 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124554016224081QQV">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3858725">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387579432806ORE">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestProfileMultipleDevices:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3858743">test_profile_change_profile_photo</a>, id: 741969 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387432540875CFB">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387165491605SPQ">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3858732">test_send_eth[Status Network Sepolia-0.0002]</a>, id: 727229 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387155355382ROR">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387225683845PJZ">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3858733">test_wallet_send_erc20_from_drawer[Optimism Sepolia-USD Coin-USDC-2-0.01]</a>, id: 727230 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387155355382ROR">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387225683845PJZ">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3858744">test_send_eth[Arbitrum Sepolia-0.0001]</a>, id: 742015 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387155355382ROR">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387225683845PJZ">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3858745">test_wallet_send_erc20_from_drawer[Sepolia-USD Coin-USDC-2-0.01]</a>, id: 742016 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387155355382ROR">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387225683845PJZ">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3858728">test_community_message_delete</a>, id: 702839 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387232866678DDK">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387435280264OKG">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3858727">test_community_message_send_check_timestamps_sender_username</a>, id: 702838 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387232866678DDK">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387435280264OKG">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3858730">test_community_one_image_send_reply_set_reaction</a>, id: 702859 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387232866678DDK">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387435280264OKG">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3858729">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387232866678DDK">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387435280264OKG">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletCollectibles:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3858740">test_wallet_send_collectible</a>, id: 741840 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387631661453HKK">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3858739">test_wallet_collectibles_balance</a>, id: 741839 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387631661453HKK">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3858741">test_wallet_collectible_send_from_expanded_info_view</a>, id: 741841 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387631661453HKK">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletOneDeviceTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3858734">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387679031209DMQ">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3858737">test_wallet_swap_flow_mainnet</a>, id: 741555 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387560528028DDO">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3858735">test_wallet_balance_mainnet</a>, id: 740490 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387560528028DDO">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3858736">test_wallet_send_flow_mainnet</a>, id: 741554 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387560528028DDO">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3858731">test_1_1_chat_send_image_save_and_share</a>, id: 703391 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387498378492BCA">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387155099446YPQ">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3858724">test_1_1_chat_pin_messages</a>, id: 702731 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387498378492BCA">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387155099446YPQ">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3858723">test_1_1_chat_message_reaction</a>, id: 702730 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387498378492BCA">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=NAAV-AND-2130125-1747124387155099446YPQ">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

--- a/src/status_im/contexts/chat/group/details/style.cljs
+++ b/src/status_im/contexts/chat/group/details/style.cljs
@@ -25,3 +25,17 @@
    :margin-bottom      bottom
    :background-color   (colors/theme-colors colors/white colors/neutral-95-opa-70 theme)
    :flex-direction     :row})
+
+(def manage-members-header
+  {:flex-direction    :row
+   :justify-content   :space-between
+   :align-items       :flex-end
+   :margin-horizontal 20
+   :margin-bottom     16})
+
+(defn counter
+  [theme error?]
+  {:margin-bottom 2
+   :color         (if error?
+                    (colors/theme-colors colors/danger-50 colors/danger-60 theme)
+                    (colors/theme-colors colors/neutral-40 colors/neutral-50 theme))})

--- a/src/status_im/contexts/chat/group/details/view.cljs
+++ b/src/status_im/contexts/chat/group/details/view.cljs
@@ -6,7 +6,6 @@
     [react-native.core :as rn]
     [react-native.gesture :as gesture]
     [react-native.safe-area :as safe-area]
-    [reagent.core :as reagent]
     [status-im.common.contact-list-item.view :as contact-list-item]
     [status-im.common.contact-list.view :as contact-list]
     [status-im.common.home.actions.view :as actions]
@@ -26,27 +25,35 @@
       (rf/dispatch [:deselect-member public-key true]))))
 
 (defn add-member-contact-item-render
-  [{:keys [public-key] :as item} _ _ {:keys [group admin? current-pk]}]
-  (let [{:keys [contacts]} group
-        member?            (contains? contacts public-key)
-        checked?           (reagent/atom member?)]
-    (if (or (= current-pk public-key) (and (not admin?) member?))
-      (fn []
-        [contact-list-item/contact-list-item
-         {:disabled? true
-          :accessory {:disabled? true
-                      :type      :checkbox
-                      :checked?  @checked?}}
-         item])
-      (fn []
-        (let [on-toggle #(group-chat-member-toggle member? (swap! checked? not) public-key)]
-          [contact-list-item/contact-list-item
-           {:on-press                on-toggle
-            :allow-multiple-presses? true
-            :accessory               {:type     :checkbox
-                                      :checked? @checked?
-                                      :on-check on-toggle}}
-           item])))))
+  [{:keys [public-key] :as item} _ _ {:keys [group admin? current-pk chat-id]}]
+  (let [{:keys [contacts]}      group
+        member?                 (contains? contacts public-key)
+        [checked? set-checked]  (rn/use-state member?)
+        selected-contacts-count (rf/sub [:group-chat/manage-members-count chat-id])
+        disabled?               (or (= current-pk public-key) (and (not admin?) member?))
+        on-toggle               (rn/use-callback
+                                 (fn [value]
+                                   (let [error? (and value
+                                                     (= constants/max-group-chat-participants
+                                                        selected-contacts-count))]
+                                     (rf/dispatch [:group-chat/set-manage-members-error
+                                                   {:chat-id chat-id
+                                                    :error?  error?}])
+                                     (when-not error?
+                                       (group-chat-member-toggle member?
+                                                                 value
+                                                                 public-key)
+                                       (set-checked value))))
+                                 [selected-contacts-count member? public-key])]
+    [contact-list-item/contact-list-item
+     {:on-press                on-toggle
+      :allow-multiple-presses? true
+      :disabled?               disabled?
+      :accessory               {:type      :checkbox
+                                :disabled? disabled?
+                                :checked?  checked?
+                                :on-check  on-toggle}}
+     item]))
 
 (defn add-manage-members
   [{:keys [on-scroll]}]
@@ -56,7 +63,9 @@
         chat-id                    (quo.context/use-screen-params)
         {:keys [admins] :as group} (rf/sub [:chats/chat-by-id chat-id])
         current-pk                 (rf/sub [:multiaccount/public-key])
-        admin?                     (get admins current-pk)]
+        admin?                     (get admins current-pk)
+        selected-contacts-count    (rf/sub [:group-chat/manage-members-count chat-id])
+        error?                     (rf/sub [:group-chat/manage-members-error-by-chat-id chat-id])]
     (rn/use-mount (fn []
                     (rf/dispatch [:group/clear-added-participants])
                     (rf/dispatch [:group/clear-removed-members])))
@@ -66,13 +75,21 @@
        :accessibility-label :close-manage-members
        :style               (style/close-icon theme)}
       [quo/icon :i/close {:color (colors/theme-colors colors/neutral-100 colors/white theme)}]]
-     [quo/text
-      {:size   :heading-1
-       :weight :semi-bold
-       :style  {:margin-left 20}}
-      (i18n/label (if admin? :t/manage-members :t/add-members))]
+     [rn/view
+      {:style style/manage-members-header}
+      [quo/text
+       {:size   :heading-1
+        :weight :semi-bold}
+       (i18n/label (if admin? :t/manage-members :t/add-members))]
+      [quo/text
+       {:size   :paragraph-2
+        :weight :regular
+        :style  (style/counter theme error?)}
+       (i18n/label :t/selected-count-from-max
+                   {:selected selected-contacts-count
+                    :max      constants/max-group-chat-participants})]]
      [gesture/section-list
-      {:key-fn                         :title
+      {:key-fn                         :public-key
        :on-scroll                      on-scroll
        :sticky-section-headers-enabled false
        :sections                       (rf/sub [:contacts/grouped-by-first-letter])
@@ -81,7 +98,8 @@
        :content-container-style        {:padding-bottom 20}
        :render-data                    {:group      group
                                         :admin?     admin?
-                                        :current-pk current-pk}
+                                        :current-pk current-pk
+                                        :chat-id    chat-id}
        :render-fn                      add-member-contact-item-render}]
      [rn/view {:style (style/bottom-container 30 theme)}
       [quo/button

--- a/src/status_im/contexts/chat/group/events.cljs
+++ b/src/status_im/contexts/chat/group/events.cljs
@@ -3,7 +3,9 @@
             [oops.core :as oops]
             [re-frame.core :as rf]
             [status-im.common.avatar-picture-picker.view :as avatar-picture-picker]
-            [taoensso.timbre :as log]))
+            [status-im.constants :as constants]
+            [taoensso.timbre :as log]
+            [utils.i18n :as i18n]))
 
 (rf/reg-event-fx :group-chat/create
  (fn [{:keys [db]} [{:keys [group-name group-color group-image]}]]
@@ -37,3 +39,16 @@
                      :on-success  (fn [response]
                                     (rf/dispatch [:chat-updated response true])
                                     (when on-success (on-success)))}]}))
+
+(rf/reg-event-fx :group-chat/set-manage-members-error
+ (fn [{:keys [db]} [{:keys [chat-id error?]}]]
+   {:db (assoc-in db
+         [:group-chat/manage-members-error chat-id]
+         error?)
+    :fx [(when error?
+           [:dispatch
+            [:toasts/upsert
+             {:type :negative
+              :text (i18n/label :t/new-group-limit
+                                {:max-contacts
+                                 constants/max-group-chat-participants})}]])]}))

--- a/src/status_im/subs/chats.cljs
+++ b/src/status_im/subs/chats.cljs
@@ -358,3 +358,37 @@
    (re-frame/subscribe [:chats/chat chat-id]))
  :->
  :image)
+
+(re-frame/reg-sub
+ :group-chat/selected-participants-count
+ :<- [:group-chat/selected-participants]
+ (fn [selected-participants]
+   (count selected-participants)))
+
+(re-frame/reg-sub
+ :group-chat/deselected-members-count
+ :<- [:group-chat/deselected-members]
+ (fn [deselected-members]
+   (count deselected-members)))
+
+(re-frame/reg-sub
+ :group-chat/member-count
+ (fn [[_ chat-id]]
+   (re-frame/subscribe [:chats/chat-by-id chat-id]))
+ (fn [{:keys [contacts]}]
+   (count contacts)))
+
+(re-frame/reg-sub
+ :group-chat/manage-members-count
+ (fn [[_ chat-id]]
+   [(re-frame/subscribe [:group-chat/member-count chat-id])
+    (re-frame/subscribe [:group-chat/selected-participants-count])
+    (re-frame/subscribe [:group-chat/deselected-members-count])])
+ (fn [[member-count selected-count deselected-count]]
+   (+ member-count selected-count (- deselected-count))))
+
+(re-frame/reg-sub
+ :group-chat/manage-members-error-by-chat-id
+ :<- [:group-chat/manage-members-error]
+ (fn [errors [_ chat-id]]
+   (get errors chat-id)))

--- a/src/status_im/subs/chats_test.cljs
+++ b/src/status_im/subs/chats_test.cljs
@@ -207,3 +207,15 @@
         :chats           chats
         :current-chat-id chat-id)
       (is (true? (rf/sub [sub-name]))))))
+
+(h/deftest-sub :group-chat/manage-members-count
+  [sub-name]
+  (testing "calculates the correct count of members in a group chat"
+    (let [contacts              [{:id "0x777"} {:id "0x789"}]
+          selected-participants ["0x123" "0x456"]
+          deselected-members    [{:id "Ox789"}]]
+      (swap! rf-db/app-db assoc
+        :chats                            {chat-id {:contacts contacts}}
+        :group-chat/selected-participants selected-participants
+        :group-chat/deselected-members    deselected-members)
+      (is (= 3 (rf/sub [sub-name chat-id]))))))

--- a/src/status_im/subs/root.cljs
+++ b/src/status_im/subs/root.cljs
@@ -129,6 +129,7 @@
 ;;group chat
 (reg-root-key-sub :group-chat/selected-participants :group-chat/selected-participants)
 (reg-root-key-sub :group-chat/deselected-members :group-chat/deselected-members)
+(reg-root-key-sub :group-chat/manage-members-error :group-chat/manage-members-error)
 
 ;;messages
 (reg-root-key-sub :messages/messages :messages)

